token-syncer/integration/token_syncer/basic_test.clj

```diff
@@ -0,0 +1,370 @@
+
+          (doall
+            (map-indexed
+              (fn [index waiter-url]
+                (waiter/store-token http-client waiter-url token-name "0"
+                                    (assoc basic-description
+                                      "cpus" (inc index)
+                                      "last-update-time" (- last-update-time-ms index)
+                                      "owner" (str "test-user-" index)
+                                      "root" "common-root")))
+              waiter-urls))
+
+          (let [token-etag (token->etag http-client (first waiter-urls) token-name)]
+
+            (log/info "****** test-token-different-owners-but-same-root ACT")
+            (let [actual-result (syncer/sync-tokens http-client waiter-urls)]
+
+              (log/info "****** test-token-different-owners-but-same-root ASSERT")
+              (let [latest-description (assoc basic-description
+                                         "cpus" 1
+                                         "last-update-time" last-update-time-ms
+                                         "owner" "test-user-0"
+                                         "root" "common-root")
+                    waiter-sync-result (constantly
+                                         {:code :success/sync-update
+                                          :details {:etag token-etag
+                                                    :status 200}})
+                    expected-result {:details {token-name {:latest {:cluster-url (first waiter-urls)
+                                                                    :description latest-description
+                                                                    :token-etag token-etag}
+                                                           :sync-result (pc/map-from-keys waiter-sync-result (rest waiter-urls))}}
+                                     :summary {:sync {:failed #{}
+                                                      :unmodified #{}
+                                                      :updated #{token-name}}
+                                               :tokens {:num-processed 1
+                                                        :total 1}}}]
+                (is (= expected-result actual-result))
+                (doall
+                  (map-indexed
+                    (fn [index waiter-url]
+                      (is (= {:description latest-description
+                              :headers {"content-type" "application/json"
+                                        "etag" token-etag}
+                              :status 200
+                              :token-etag token-etag}
+                             (waiter/load-token http-client waiter-url token-name))))
+                    waiter-urls))))))
+        (finally
+          (cleanup-token http-client waiter-urls token-name))))))
+
+(deftest ^:integration test-token-different-owners-and-roots
```

Comment:
Are owners relevant?  Is it enough to say that different roots should never be merged?